### ISSUE TITLE
Improve Source List Organization / Metadata

### ISSUE THREAD

**Issue by cmelbye** [jaedb/Iris#690]
**Is your feature request related to a problem? Please describe.**
Snapcast integration in Iris is really great, but the source list doesn't convey as much information as it could. It isn't easy to see which rooms are currently synchronized and what they're listening to. (Just the name of the source)

<img width="886" alt="existing-source-list" src="https://user-images.githubusercontent.com/11472/106829417-5636a680-6649-11eb-91b0-4941afc30b6b.png">

**Describe the solution you'd like**
* The source list could be **organized by what you are listening to**, not where you can listen.
* It could also incorporate metadata provided by Snapcast. This would give a **better picture of the audio that's currently playing on a source**, or whether the source is playing anything at all.

This is just an idea of potential UX to start a conversation:

<img width="886" alt="source-list" src="https://user-images.githubusercontent.com/11472/106829973-6307ca00-664a-11eb-934c-e4526f5a1096.png">

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by jaedb** [comment]
I can see you've put a decent amount of thought into how this could function - I really like where you're going with this. As I understand it, Snapcast doesn't provide [metadata of the source audio](https://github.com/badaix/snapcast/blob/master/doc/json_rpc_api/v2_0_0.md#stream) which would make this somewhat difficult to achieve.

It looks like there is an existing PR for Snapcast to pull this data: https://github.com/badaix/snapcast/issues/749

In _theory_ Iris could lean on the _Settings > Servers_ section and maintain simultaneous connections to all of those servers to get the meta from Mopidy directly, however I wouldn't expect it to be reliable in matching Mopidy servers to Snapcast streams.

**Comment by cmelbye** [comment]
@jaedb Yes, currently Snapcast does support metadata for certain stream types. The only one I'm aware of is AirPlay/shairport-sync but there could be others. I can see that Iris sees track, artist, cover art, etc. in its debug request to Snapcast.

And it looks like they plan to add support for MPD soon, too. For streams that don't have metadata, there would always need to be some sort of fallback in Iris.

**Comment by cmelbye** [comment]
This is the response to `Server.GetStatus` when I'm using AirPlay:

{
  "server": {
    [removed for brevity]
    "streams": [
      [removed for brevity]
      {
        "id": "AirPlay",
        "meta": {
          "ALBUM": "All That - Single",
          "ARTIST": "Emotional Oranges & Channel Tres",
          "COVER": "[removed for brevity]",
          "STREAM": "AirPlay",
          "TITLE": "All That"
        },
        "status": "playing",
        "uri": {
          "fragment": "",
          "host": "",
          "path": "/home/pi/shairplay-sync",
          "query": {
            "chunk_ms": "20",
            "codec": "flac",
            "name": "AirPlay",
            "sampleformat": "44100:16:2"
          },
          "raw": "airplay:////home/pi/shairplay-sync?chunk_ms=20&codec=flac&name=AirPlay&sampleformat=44100:16:2",
          "scheme": "airplay"
        }
      }
    ]
  }
}

**Comment by jaedb** [comment]
Ah, that is interesting. The examples and (at the time) recommended implementation of Mopidy + Snapcast was to use `filesink`, which appears to omit this level of metadata. This is also a simple method of getting audio streaming between docker containers (which Iris encourages) so is what I've used to date.

I feel that when Snapcast provides support for MPD, then this would be a lot more straightforward.

Have you managed any other simple method of getting audio between docker containers? Check out [`docker-compose.example.yml`](https://github.com/jaedb/Iris/blob/master/docker-compose.example.yml) for the current approach. I had experimented with pulseaudio and it was a terrible experience.

**Comment by cmelbye** [comment]
Yeah, until Snapcast provides support for this, I don't think it's possible. The existing support for metatags is very specific to AirPlay. This WIP branch is adding support for MPD: https://github.com/badaix/snapcast/compare/metatags#diff-dff401eacf0622e6c71058ab8bc65e87bb19470fe5801fc0a086cdffbfc55f20R39

From what I can tell, it is using the same `pipe` stream that the Iris docker configuration is using. In the branch above, that IP/port is currently hardcoded to `127.0.0.1:6600` but I imagine it would be a configuration option when finalized.

So perhaps that docker example would need to inform the `snapserver` container of an IP/port from the `mopidy` container that it can use for MPD connection. I forget how services work in Docker but I assume this is possible to set up...

**Comment by cmelbye** [comment]
(But of course, Iris already knows what Mopidy is playing even without this change, so there could instead be a way to identify that a particular stream on snapserver corresponds to the audio that the Mopidy instance is playing. I might be missing something that would make this approach less optimal, though.)

**Comment by bendschs** [comment]
+1 for that, snapcast is getting metadata from shairport sync already.
